### PR TITLE
ESC-230 fix compiler warnings 

### PR DESCRIPTION
**Pull request by carwynellis**
Summary of changes
* added scoverage plugin with no coverage limit implemented yet
* resolved compiler warnings and made some minor tidy ups

### CODE REVIEW

File: app/uk/gov/hmrc/eusubsidycompliance/controllers/MicroserviceHelloWorldController.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ import scala.concurrent.Future
 class MicroserviceHelloWorldController @Inject()(cc: ControllerComponents)
     extends BackendController(cc) {
 
-  def hello(): Action[AnyContent] = Action.async { implicit request =>
+  def hello(): Action[AnyContent] = Action.async { _ =>
```
Comment (priyankadhiman3006) [inline app/uk/gov/hmrc/eusubsidycompliance/controllers/MicroserviceHelloWorldController.scala:28]: Shall we remove this?

**Comment by priyankadhiman3006** [comment]
While you are doing this, shall we include a scalafmt file as well to adjust the formatting? what do you guys think?

**Review by priyankadhiman3006 — APPROVED** [review]

File: app/uk/gov/hmrc/eusubsidycompliance/controllers/MicroserviceHelloWorldController.scala (hunk 1 above)
Comment (carwynellis) [reply app/uk/gov/hmrc/eusubsidycompliance/controllers/MicroserviceHelloWorldController.scala:28]: It's being removed in the next PR. This just deals with the immediate warnings for now.

**Comment by carwynellis** [comment]
> While you are doing this, shall we include a scalafmt file as well to adjust the formatting? what do you guys think?

Hmm...we can sort this later. Is there a standard HMRC config that we can use?

**Comment by priyankadhiman3006** [comment]
> > While you are doing this, shall we include a scalafmt file as well to adjust the formatting? what do you guys think?
> 
> Hmm...we can sort this later. Is there a standard HMRC config that we can use?

yeah sure, I can share what we used in my other project and then we can  review and decide on that

**Review by sam-miele — APPROVED** [review]

**Pull request merged**